internal/ice/util.go

```diff
@@ -38,8 +38,31 @@ func localInterfaces() (ips []net.IP) {
 				continue
 			}
 
+			var IPv4Supported, IPv6Supported bool
+			for _, typ := range networkTypes {
+				if typ.IsIPv4() {
+					IPv4Supported = true
+				}
+
+				if typ.IsIPv6() {
+					IPv6Supported = true
+				}
+			}
+
 			// The conditions of invalidation written below are defined in
 			// https://tools.ietf.org/html/rfc8445#section-5.1.1.1
+			if !IPv4Supported && !IPv6Supported {
```

Comment:
I worked out the logic below with a binary conditional table. This still maintains the logic you wanted with the last line being `ips = append(ips, ip)` but all the other statements work through eliminating the ips that do not fit the supported passed list.